.github/workflows/audit.yml

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Security audit
+
+on:
+  schedule:
+    # https://crontab.guru/#0_0_*_*_3
+    # “At 00:00 on Wednesday.”
+    - cron: "0 0 * * *"
+
+jobs:
+  audit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/audit-check@v1
```

Comment:
https://github.com/marketplace/actions/rust-audit-check

Reply:
I think we are trying to avoid using github actions other than "official" ones (aka `actions` is good, but `action-rs` is not as it is not provided by github: https://github.com/actions-rs)

Though I am just basically cargo culting this line (aka repeating back what I have heard in the past) without really knowing how important/accurate it is

Reply:
we can let this part for a while.

i also heard the same, but only thinking that such actions shall not acquire github tokens so it changes code (i.e. due to IP constraints). this action shall only scans the code without writing to it.

i'm okay with not merging it as well.

Reply:
I think the issue is that that the `GITHUB_TOKEN` is conected to the token of the user who created the PR -- thus it may have non trivial credentials and if someone gained control of actions-rs they could then replace `actions-rs/audit-check@v1` with some code that abused those credentials

Reply:
We can instead directly use the `cargo audit` command?

Reply:
I think we can do that but the problem is we cannot let it block CI

Sometimes auditing failures requires longer resolution time￼￼